HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,16 @@
+import pyspark
+from pyspark import SparkContext, SparkConf
+#sc = SparkContext()
+
+def bfs(graph, starting_char):
+    char_list = [starting_char]
+    BFS_list  = []
+    
+    while len(char_list)!=0:
+        neighbors = graph.filter(lambda tpl: tpl[0] in char_list).flatMap(lambda tpl: tpl[1]).collect()
+        BFS_list  = BFS_list + char_list
+        char_list = []
+        char_list = char_list + list(set(neighbors)-set(BFS_list))                                     
+        print len(list(set(neighbors)-set(BFS_list)))
```

Comment:
I agree with "araftery". The BFS you implemented here is very close to the original version. There is a specifically designed parallelized BFS algorithm which you could found online. The basic idea is mentioned in the homework document, update the distance each loop.